clippy_lints/src/no_effect.rs

```diff
@@ -35,6 +35,10 @@ declare_clippy_lint! {
     /// Expressions by themselves often have no side-effects.
     /// Having such expressions reduces readability.
     ///
+    /// ### Known problems
+    /// Suggested improvement can lead to semantic change in certain edge cases, such as use of
+    /// function call in indexing to conditionally cause panic.
```

Comment:
Please add an example code snippet here (doesn't have to be a code block). Also, please add that if this is intentional, this lint should be allowed.